sixtools/info_functions.py

```diff
@@ -0,0 +1,94 @@
+
+
+        if source == 'all' or source == 'header':
+            f_string+='HEADER INFORMATION\n------------------\n'
+            f_string+='    START INFORMATION\n-----------------\n'
+            for key in list(hdr.start.keys()): 
+                if key == 'time':
+                    f_string+='        time'.ljust(20)+' :'.ljust(4)+str(datetime.datetime.fromtimestamp(hdr.start['time']).strftime('%c'))+'\n'
+                elif key != 'scan_id' and key != 'plan_args':
+                    f_string+='        '+key.ljust(20)+' :'.ljust(4)+str(hdr.start[key])+'\n'
+
+            f_string+='    STOP INFORMATION\n-----------------\n'
+            for key in list(hdr.stop.keys()): 
+                if key == 'time':
+                    f_string+='        time'.ljust(20)+' :'.ljust(4)+str(datetime.datetime.fromtimestamp(hdr.stop['time']).strftime('%c'))+'\n'
+                elif key != 'scan_id' and key != 'plan_args':
+                    f_string+='        '+key.ljust(20)+' :'.ljust(4)+str(hdr.stop[key])+'\n'
+        
+        if source == 'all' or source == 'baseline':
+            BL = hdr.table(stream_name='baseline')
+            f_string+='BASELINE INFORMATION\n------------------\n'
+            
+            exit_val=0
+            
+            keys = list(BL.keys())
+            while len(keys) > 0 and exit_val <=50:
+                exit_val+=1
+                if keys[0] == 'time':
+                    device = keys[0]
+                else:
+                    device = keys[0].partition('_')[0]
+                f_string+='    '+str(device)+'\n-----------\n'
+                device_keys = list(key for key in keys if key.startswith(device))
+                for key in device_keys:
+
+                    f_string+='        '+key.ljust(30)+' start val : '+str(BL[key][1]).ljust(30)+' stop val : '+str(BL[key][2])+'\n'
+
+                keys = list(key for key in keys if not key.startswith(device))
+                    
+        print (f_string)
+  
+    else:
+        print ('source must be "all", "header" or "baseline"')
+
+
+
+### detector utilities ##
+
+def scan_dets(*scan_no):
+    '''Enter scan id numbers to print settings for primary stream detectors'''
+   
+    
+    xtr_str = ''
+    for scan in scan_no:
+        if 'reason' in db[scan].start:
+            xtr_str = '   ' + db[scan].start['reason']
+        elif 'purpose' in db[scan].start:
+            xtr_str = '   ' + db[scan].start['purpose']
+        else:
+            xtr_str = '   '
+            print('Scan ID {}:  {}'.format(scan, xtr_str) )
+
+        for detector in db[scan]['start']['detectors']:
+            print('  {} (primary stream):  '.format(detector) )
+            settings = db[scan].config_data(detector)['primary'][0]
+                
+            try:
+                for p, n in settings.items():
+                    print('    {:_<30} : {:_>20}'.format(p, n))
+                print('{:-<70}'.format('-'))
+            except:
+                pass
```

Comment:
I am not sure I like the idea of passing at this point. To me it means we are not providing all the information from the config attributes and this may get confusing. Can we just not format n in this part of the code?

Reply:
@awalter-bnl I think that all the other information that prints out for our use case is not helpful.  I can add an option for `verbose = False` to automatically pass on lists for now if you feel like it is important to let some access this information in an easy way, but I don't want to see it myself as it is not helpful.  

Perhaps, in the future, easily recognized dictionary keys for this excess information can be configured so that it is easy to print or pass these things.  See also https://github.com/NSLS-II/ophyd/issues/587 and https://github.com/NSLS-II/ophyd/issues/586

Reply:
I don't feel really strongly on this to be honest. My conern is that the easiest way to describe what this does is *it prints, in human readable form, the configuration attributes for the primary detectors*. However if we are passing on the lists etc. then we are losing some of that info. If you feel strongly enough then we can leave it as is (it is after all a six beamline function).